substrate/frame/babe/src/lib.rs

```diff
@@ -25,26 +25,20 @@ extern crate alloc;
 
 use alloc::{boxed::Box, vec, vec::Vec};
 use codec::{Decode, Encode};
-use frame_support::{
-	dispatch::{DispatchResultWithPostInfo, Pays},
-	ensure,
-	traits::{ConstU32, DisabledValidators, FindAuthor, Get, OnTimestampSet, OneSessionHandler},
-	weights::Weight,
-	BoundedVec, WeakBoundedVec,
+use frame::{
+	deps::sp_runtime::generic::DigestItem,
+	prelude::*,
+	traits::{
+		DisabledValidators, EstimateNextSessionRotation, FindAuthor, IsMember,
```

Comment:
A few of these seems general enough to go to prelude: 

`IsMember`, `FindAuthor`, `OnTimestampSet`. 

Rule of thumb: If you can foresee it being useful in any pallet, then prelude. Else, domain specifc mod (e.g. `mod traits`).

Reply:
Perfect, understood.